private/bufpkg/bufmodule/bufmodulebuild/module_file_set_builder.go

```diff
@@ -121,7 +123,6 @@ func (m *moduleFileSetBuilder) build(
 			}
 			if _, ok := hashes[potentialDependencyModuleHash]; !ok {
 				dependencyModules = append(dependencyModules, potentialDependencyModule)
-			} else {
 				hashes[potentialDependencyModuleHash] = struct{}{}
 			}
```

Comment:
This seems like something that we missed [here](https://github.com/bufbuild/buf/pull/2232/files#diff-b267002b1d5ca9c7c22a25c7f3016a781a0322ba50822036c0e6aebf3f2d9fd8R122-R126). `hashes` is of type `map[string]struct{}`, doesn't make sense to add a key to the map when it's already there. Probably we wanted to append to `dependencyModules` and also mark it as present in the `hashes` map, but I'm not 100% sure. Pls confirm @bufdev

Reply:
yea this seems like a mistake on my part, good catch